src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -121,4 +126,61 @@ public static Tag parseTag(String tag) throws ParseException {
         }
         return tagSet;
     }
+
+    /**
+     * Parses a {@code String author} into a {@code Author}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code author} is invalid.
+     */
+    public static Author parseAuthor(String author) throws ParseException {
+        requireNonNull(author);
+        String trimmedAuthor = author.trim();
+        if (!Author.isValidAuthor(trimmedAuthor)) {
+            throw new ParseException(Author.MESSAGE_CONSTRAINTS);
+        }
+        return new Author(trimmedAuthor);
+    }
+
+    /**
+     * Parses {@code Collection<String> authors} into a {@code List<Author>}.
+     */
+    public static List<Author> parseAuthors(Collection<String> authors) throws ParseException {
+        requireNonNull(authors);
+        final List<Author> authorList = new ArrayList<>();
+        for (String authorName : authors) {
+            authorList.add(parseAuthor(authorName));
+        }
+        return authorList;
+    }
+
+    /**
+     * Parses a {@code String bookName} into a {@code BookName}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code bookName} is invalid.
+     */
+    public static BookName parseBookName(String bookName) throws ParseException {
+        requireNonNull(bookName);
+        String trimmedBookName = bookName.trim();
+        if (!BookName.isValidBookName(trimmedBookName)) {
+            throw new ParseException(BookName.MESSAGE_CONSTRAINTS);
+        }
+        return new BookName(trimmedBookName);
+    }
+
+    /**
+     * Parses a {@code String isbn} into a {@code Isbn}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code isbn} is invalid.
+     */
+    public static Isbn parseIsbn(String isbn) throws ParseException {
+        requireNonNull(isbn);
+        String trimmedIsbn = isbn.trim();
+        if (!Isbn.isValidIsbn(trimmedIsbn)) {
+            throw new ParseException(Isbn.MESSAGE_CONSTRAINTS);
+        }
+        return new Isbn(trimmedIsbn);
+    }
```

Comment:
I have written similar functions as well, but will still proceed to merge yours first and resolve the merge conflicts on my own local fork.